frontend/test/metabase/components/TokenField.unit.spec.js

```diff
@@ -195,118 +188,122 @@ describe("TokenField", () => {
-      focusAndType("G");
-      expect(options()).toEqual(["Gadget", "Gizmo"]);
-      keyDown(KEYCODE_ENTER);
-      expect(options()).toEqual(["Gizmo"]);
-      focus();
-      expect(options()).toEqual(["Doohickey", "Gizmo", "Widget"]);
+      fireEvent.change(input(), { target: { value: "G" } });
+      within(options()).getByText("Gadget");
+      within(options()).getByText("Gizmo");
+
+      fireEvent.keyDown(input(), { keyCode: KEYCODE_ENTER });
+      within(options()).getByText("Gizmo");
+      expect(within(options()).queryByText("Doohickey")).toBeNull();
+      expect(within(options()).queryByText("Widget")).toBeNull();
+      expect(input().value).toEqual("");
+
+      // Reset search on focus (it was a separate test before)
+      input().focus();
+      within(options()).getByText("Doohickey");
+      within(options()).getByText("Widget");
     });
 
     it("should reset the search when adding the last option", () => {
-      focusAndType("G");
-      expect(options()).toEqual(["Gadget", "Gizmo"]);
-      keyDown(KEYCODE_ENTER);
-      expect(options()).toEqual(["Gizmo"]);
-      keyDown(KEYCODE_ENTER);
-      expect(options()).toEqual(["Doohickey", "Widget"]);
+      fireEvent.change(input(), { target: { value: "G" } });
+      within(options()).getByText("Gadget");
+      within(options()).getByText("Gizmo");
+
+      fireEvent.keyDown(input(), { keyCode: KEYCODE_ENTER });
+      within(options()).getByText("Gizmo");
+
+      fireEvent.keyDown(input(), { keyCode: KEYCODE_ENTER });
+      within(options()).getByText("Doohickey");
+      within(options()).getByText("Widget");
     });
 
     it("should hide the option if typed exactly then press enter", () => {
-      focusAndType("Gadget");
-      expect(options()).toEqual(["Gadget"]);
-      keyDown(KEYCODE_ENTER);
-      expect(values()).toEqual(["Gadget"]);
-      expect(options()).toEqual(["Doohickey", "Gizmo", "Widget"]);
+      fireEvent.change(input(), { target: { value: "Gadget" } });
+      within(options()).getByText("Gadget");
+      fireEvent.keyDown(input(), { keyCode: KEYCODE_ENTER });
+      within(values()).getByText("Gadget");
+
+      within(options()).getByText("Doohickey");
+      within(options()).getByText("Gizmo");
+      within(options()).getByText("Widget");
     });
 
     it("should hide the option if typed partially then press enter", () => {
-      focusAndType("Gad");
-      expect(options()).toEqual(["Gadget"]);
-      keyDown(KEYCODE_ENTER);
-      expect(values()).toEqual(["Gadget"]);
-      expect(options()).toEqual(["Doohickey", "Gizmo", "Widget"]);
+      fireEvent.change(input(), { target: { value: "Gad" } });
+      within(options()).getByText("Gadget");
+      fireEvent.keyDown(input(), { keyCode: KEYCODE_ENTER });
+      within(values()).getByText("Gadget");
+      within(options()).getByText("Doohickey");
+      within(options()).getByText("Gizmo");
+      within(options()).getByText("Widget");
```

Comment:
Here, it's kinda nice that the old assertions are more specific. 

`expect(options()).toEqual(["Doohickey", "Gizmo", "Widget"]);` also asserts that "Gadget" isn't in the list. Would it be possible to add a helper that also checks how many items are in the list? 
e.g. `optionsContains(["Doohickey", "Gizmo", "Widget"])`

Maybe looking at something like `within(options()).getAllByText(/./).length`?

Reply:
I'll see what can I do about it. There must be some way to check this. What I know for sure is that we can check for `textContent` but it concatenates all text content. In other words, if you want to check if options contain `["Doohickey", "Gizmo", "Widget"]`, the text content would give you `DoohickeyGizmoWidget`. If you're ok with it, I can use that approach?

Reply:
👍 Whatever works! The helper will only be used in this file so I don't think it needs to be elegant as long as it asserts the right thing.

Reply:
Added helpers in 2a05e29158b096514d353ab1a4b161f7aff0b403. Not only did it help to write more precise and succinct code, but it also removed the need to check for non-existence of some elements.